python/pyarrow/types.pxi

```diff
@@ -902,6 +902,19 @@ cdef class ExtensionType(BaseExtensionType):
         """
         return ExtensionArray
 
+    def scalar_as_py(self, scalar):
+        """Convert scalar to a Python type.
+
+        This method can be overridden in subclasses to customize what type
+        scalars are converted to. Implementations should handle the case
+        where the scalar argument is ``None``.
```

Comment:
I am curious about your intention to leave the `None` handling to the subclasses.

Would you like to allow returning custom values (along the lines of `numpy.NaN`) in the case that the value is `None`? Or should `None` always stay `None` because Arrow actually is more explicit with its missing value handling and is able to differentiate between e.g. `None` and `NaN` for double columns. If the latter is preferred, then I'd suggest to move the `None` handling to `Scalar.as_py()` and only invoke `scalar_as_py` for existing values.

Reply:
I think for numpy conversion, we have more efficient methods to do that conversion (to arrays). I can't think of any other case where someone might want to customize the null handling of `as_py()`, so maybe you are right that we should only invoke `scalar_as_py` for existing values.

@jorisvandenbossche Any thoughts on that?